metricbeat/helper/sql/sql.go

```diff
@@ -0,0 +1,205 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
+
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type DbClient struct {
+	*sql.DB
+	logger *logp.Logger
+}
+
+type sqlRow interface {
+	Scan(dest ...interface{}) error
+	Next() bool
+	Columns() ([]string, error)
+	Err() error
+}
+
+// DB gets a client ready to query the database
+func DB(driver, uri string, l *logp.Logger) (*DbClient, error) {
```

Comment:
Since we changed the struct's name, should we also tune the getter according to https://golang.org/doc/effective_go.html#Getters? :)

Reply:
Actually this is more a constructor than a getter, isn't it? https://golang.org/doc/effective_go.html#composite_literals :slightly_smiling_face: :slightly_smiling_face:

Reply:
Yeah, worth mentioning that even small changes in code or naming can cause chain-reactions in many other places and this is a good example of it

Reply:
Isn't refactoring/renaming of functions/variables handled by editors quite smoothly? In any case if it's about to cause other issues we can keep it as is!

Reply:
Oh yes, I'd love that IDE's take care of **proper naming** (not renaming / refactoring) for us :smile: But I think that AI hasn't reached that level yet :wink:

Reply:
I left it with `type DbClient` and `func NewDbClient`